proxy/http/ConnectingEntry.cc

```diff
@@ -152,13 +152,12 @@ ConnectingEntry::state_http_server_open(int event, void *data)
 void
 ConnectingEntry::remove_entry()
 {
-  EThread *ethread = this_ethread();
-  auto ip_iter     = ethread->connecting_pool->m_ip_pool.find(this->ipaddr);
-  while (ip_iter != ethread->connecting_pool->m_ip_pool.end() && this->ipaddr == ip_iter->first) {
+  EThread *ethread            = this_ethread();
+  auto [iter_start, iter_end] = ethread->connecting_pool->m_ip_pool.equal_range(this->ipaddr);
```

Comment:
So using equal_range is the key to the fix?  equal_range will use the IpHelper compare operator that now includes the port and the IP address.  The original find() will also use the ip address and port, but the ipaddr comparison in the while loop will not.

Assuming I'm understanding that correctly, the changes look good to me.

Reply:
Yes, you are correct. Using equal_range uses the new IpHelper comparison which makes selection consistent across the various uses of ip pool.